R/ttestpairedsamples.R

```diff
@@ -436,25 +436,26 @@ TTestPairedSamples <- function(jaspResults, dataset = NULL, options, ...) {
   data <- data.frame(id = rep(1:length(c1), 2), dependent = c(c1, c2),
                      groupingVariable = c(rep(paste("1.", pair[[1]], sep = ""), length(c1)),
                                           rep(paste("2.", pair[[2]], sep = ""), length(c2))))
-  
+
   summaryStat <- summarySEwithin(data, measurevar = "dependent", withinvars = "groupingVariable",
                                  idvar = "id", conf.interval = options$descriptivesPlotsConfidenceInterval,
                                  na.rm = TRUE, .drop = FALSE)
-  
+  summaryStat[["groupingVariable"]] <- factor(summaryStat[["groupingVariable"]])
```

Comment:
Suggestion: fix this inside `summarySEwithin()` to ensure other code that uses this function behaves as it did previously

Reply:
I've done this now but I'm not sure if this is a good idea. Now, it *might* happen that `$betweenvars` is a character and `$withinvars` is a factor. Since these functions are also used in the ANOVAs, I'm a bit hesitant about modifying them too much (since the unit tests here won't catch any error there). These functions actually deserve some extensive unit tests of their own before we can safely refactor them.

Reply:
I've not looked into the function whatsoever, I only figured that previously it returned a factor and all receiving parties were happy, so it's probably good for it to continue returning factors (especially if it seems to be a better data type for subsequent analysis).

Reply:
Yeah that works for me.
Can you check if the ANOVA's still use this function correctly? It might just be we're not testing it there.